### DISCUSSION TITLE
Passing variable from automation to script errors

### DISCUSSION THREAD

**Discussion by pjdreger3** [music-assistant/support#1487] [Q&A] [answered]
I am using the mass.queue service to play spotify because the implementation requires the ability to clear the queue when ever a trigger event occurs (the implementation is a experience room for autistic grandson and using NFC tags to allow hom to kick of an event - he will start and stop events all the time). I wasnt able to determine how to clear the queue using the play_media but know how to do it with mass.queue.

All that to say I have an automation that defines the variables for each tag  as follows:
```
id: DancePartyA
platform: tag
tag_id: 04-8B-CF-36-B8-2A-81
variables:
c_type: audio
pl_type: Kids Rock! - Rock Music for Children
m_type: null
l_type: DanceParty
```
and then a set of conditions that kick off a script based on the c_type variable. THe conditional code follows:
action:
```
`service:` persistent_notification.create
data:
title: Debug Notification
message: “Content Type: {{ trigger.c_type }}”
choose:
conditions:
condition: template
value_template: “{{trigger.c_type == audio }}”
sequence:
service: script.1692643185235
data:
playlist: “{{trigger.variables.pl_type}}”
conditions:
condition: template
value_template: “{{trigger.c_type == video_s }}”
sequence:
service: script.1692643378131
data:
media: “{{trigger.variables.m_type}}”
conditions:
condition: template
value_template: “{{trigger.variables.c_type == video_p}}”
sequence:
service: script.1692643035619
data:
media: “{{trigger.variables.m_type}}”
playlist: “{{trigger.variables.pl_type}}”
mode: `single`
```
the script that is kicked off by the audio c_type variable is:
```
alias: PlaySpotifyPL
sequence:

service: mass.queue_command
data:
command: play_media
enqueue_mode: play
uri: “{{playlist}}”
player_id: media_player.wcast_tv_3
mode: single
```
My problem seems to be the variable isn't being passed in a format the mass.queue service accepts.  If I hardcode a playlist name in the script everything works (but is not an acceptable solution for maintainability )

I don't think this is a MASS issue but thought I'd try here in case there is some restriction on MASS accepting passed variables . I've also  posted in in HASS Forum (https://community.home-assistant.io/t/pass-variable-from-automation-to-script-not-working/605506) because I've been fooling around with this for 3 days with no luck.

Any help greatly appreciated

**Comment by OzGav** [comment]
Any chance you can fix the indenting so the yaml is easier to read?

**Reply by pjdreger3** [reply]
I used the ``` before and after what am I doing wrong

**Reply by OzGav** [reply]
Remove them all and use the <> button to do the formatting. Looks like there are extra quotes in there…?

edit: and when I go to edit your post there aren’t any indents so copy and paste again from the source

**Reply by pjdreger3** [reply]
The NFC Trigger definition

`  - id: GranA
    platform: tag
    tag_id: 04-E4-A5-37-B8-2A-81
    variables:
      c_type: audio
      pl_type: Gran
      m_type: null
      l_type: Gran
`

The conditions that react to the trigger

`action:
  - service: persistent_notification.create
    data:
      title: Debug Notification
      message: "Content Type: {{ trigger.c_type }}"
  - choose:
      - conditions:
          - condition: template
            value_template: "{{trigger.c_type == audio }}"
        sequence:
          - service: script.1692643185235
            data:
              playlist: pl_type
      - conditions:
          - condition: template
            value_template: "{{trigger.c_type == video_s }}"
        sequence:
          - service: script.1692643378131
            data:
              media: "{{trigger.variables.m_type}}"
      - conditions:
          - condition: template
            value_template: "{{trigger.variables.c_type == video_p}}"
        sequence:
          - service: script.1692643035619
            data:
              media: "{{trigger.variables.m_type}}"
              playlist: "{{trigger.variables.pl_type}}"
mode: single
`
THe test case is following the audio branch which calls the script to play the playlist (pl_type) called Gran

The PlaySpotifyPlaylist script (called when c_type = audio

`alias: PlaySpotifyPL
sequence:
  - service: mass.queue_command
    data:
      command: play_media
      enqueue_mode: play
      uri: playlist
      player_id: media_player.wcast_tv_3
mode: single
`
Nothing I've tried on either the calling side (automation) or the called side (script) results in the playlist named "Gran" to be passed to the script. I've seen it pass Playlist, pl_type etc . 

Right now Im stuck

**Comment by pjdreger3** [comment]
for some reason the trigger definition didnt format right
`  - id: GranA
    platform: tag
    tag_id: 04-E4-A5-37-B8-2A-81
    variables:
      c_type: audio
      pl_type: Gran
      m_type: null
      l_type: Gran
`

**Reply by OzGav** [reply, marked as answer]
https://community.home-assistant.io/t/pass-variable-from-automation-to-script-not-working/605506/4

**Comment by krisnoble** [comment]
> I wasnt able to determine how to clear the queue using the `play_media`

Does `enqueue: replace` not do what you want? For me it plays the requested media and clears the queue which is what it seems like you're doing with the `mass.queue`?
